game-engine/src/physics/collision.rs

```diff
@@ -0,0 +1,376 @@
+        },
+        Collider::BoxColider(b1) => match c2 {
+            Collider::SphereColider(b2) => {
+                collide_sphere_vs_box(b2, rb2, t2, w2, b1, rb1, t1, w1)
+            }
+            Collider::BoxColider(b2) => collide_box_vs_box(b1, rb1, t1, w1, b2, rb2, t2, w2),
+        },
+    }
+}
+
+pub fn collide(
+    rb1: &mut RidgidBody,
+    t1: &mut Transform,
+    cc1: &Vec<Collider>,
+    t2: &mut Transform,
+    rb2: &mut RidgidBody,
+    cc2: &Vec<Collider>,
+) -> bool {
+    let mut has_colided = false;
+
+    for c1 in cc1 {
+        for c2 in cc2 {
+            if is_colliding(c1, t1, c2, t2) {
+                has_colided = true;
+
+                let rot_1 = Mat3::from(t1.rotation);
+                let rot_2 = Mat3::from(t2.rotation);
+
+                let i_1 = rot_1 * c1.inv_inertia_tensor() * rot_1.transposed();
+                let i_2 = rot_2 * c2.inv_inertia_tensor() * rot_2.transposed();
+
+                debug_assert_finite!(rb1.velocity());
+                debug_assert_finite!(rb2.velocity());
+                debug_assert_finite!(rb1.angular_velocity(i_1));
+                debug_assert_finite!(rb2.angular_velocity(i_2));
+
+                //rb1.is_active = true;
+                //rb2.is_active = true;
+                rb1.is_active_time = 0.0;
+                rb2.is_active_time = 0.0;
+
+                solve_colliding(c1, rb1, t1, c2, rb2, t2);
+            }
+        }
+    }
+
+    rb1.is_colliding_this_frame = has_colided || rb1.is_colliding_this_frame;
+    rb2.is_colliding_this_frame = has_colided || rb2.is_colliding_this_frame;
+
+    has_colided
+}
+
+impl RidgidBody {
+    pub fn add_impulse(&mut self, force: Vec3) {
+        self.linear_momentum += force;
+    }
+
+    pub fn step(&mut self, dt: f32, transform: &mut Transform, inv_inertia_tensor: Mat3) {
+        if self.is_static {
+            return;
+        }
+
+        debug_assert_finite!(self.velocity());
+
+        //TODO https://en.wikipedia.org/wiki/Verlet_integration
+
+        transform.position += self.velocity() * dt; 
+        debug_assert_finite!(transform.position);
+
+        self.add_impulse(self.acceleration * dt * self.mass);
+
+        // apply rotation
+        let i_inv = Mat3::from(transform.rotation)
+            * inv_inertia_tensor
+            * Mat3::from(transform.rotation).transposed();
+        let angular_velocity = self.angular_velocity(i_inv);
+
+        transform.rotation.rotate_x(angular_velocity.x * dt);
+        transform.rotation.rotate_y(angular_velocity.y * dt);
+        transform.rotation.rotate_z(angular_velocity.z * dt);
+
+        self.is_active_time += dt;
+    }
+}
+
+pub fn update(
+    is_paused: &mut bool,
+    dt: f32,
+    transforms: &mut Vec<Transform>,
+    phx_objects: &mut Vec<PhysicsObject>,
+) {
+    let real_dt = dt;
+    let phx_length = phx_objects.len();
+
+    for i in 0..phx_length {
```

Comment:
Would be more idiomatic to iterate over `phx_objects.iter_mut().zip(transforms.iter_mut())`, but this will be changed when integrating with the ECS anyways